tests/plugins/test_reactor_config.py

```diff
@@ -0,0 +1,224 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from jsonschema import ValidationError
+import io
+import logging
+import os
+import pkg_resources
+import pytest
+from textwrap import dedent
+import re
+import yaml
+
+from atomic_reactor.core import DockerTasker
+from atomic_reactor.inner import DockerBuildWorkflow
+from atomic_reactor.plugins.pre_reactor_config import (DEFAULT_CONFIG,
+                                                       ReactorConfig,
+                                                       ReactorConfigPlugin,
+                                                       get_config)
+from tests.constants import TEST_IMAGE
+from tests.docker_mock import mock_docker
+from flexmock import flexmock
+
+
+class TestReactorConfigPlugin(object):
+    def prepare(self):
+        mock_docker()
+        tasker = DockerTasker()
+        workflow = DockerBuildWorkflow({'provider': 'git', 'uri': 'asd'},
+                                       TEST_IMAGE)
+        return tasker, workflow
+
+    def test_no_config(self):
+        tasker, workflow = self.prepare()
+        conf = get_config(workflow)
+        assert isinstance(conf, ReactorConfig)
+
+        same_conf = get_config(workflow)
+        assert conf is same_conf
+
+    def test_no_schema_resource(self, tmpdir, caplog):
+        class FakeProvider(object):
+            def get_resource_stream(self, pkg, rsc):
+                raise IOError
+
+        (flexmock(pkg_resources)
+            .should_receive('get_provider')
```

Comment:
It's a class instance that it gets. I could add `.with_args(object)`? This is pkg_resources internals really, but it won't let me flexmock `get_resource_stream()` which is all I care about mocking.